HPXMLtoOpenStudio/resources/hpxml.rb

```diff
@@ -966,6 +967,13 @@ def to_oga(doc)
           XMLHelper.add_element(extension, 'SchedulesFilePath', schedules_filepath, :string)
         end
       end
+      if (not @extension_properties.nil?)
```

Comment:
Need to check for empty hash too, so as not to produce a AdditionalProperties element with nothing inside.